addons/l10n_in_ewaybill/models/l10n_in_ewaybill.py

```diff
@@ -0,0 +1,599 @@
+        ("I", "Inward")
+    ], compute='_compute_supply_type')
+    partner_bill_from_id = fields.Many2one(
+        'res.partner',
+        string='Bill From',
+        compute='_compute_document_partners_details',
+        check_company=True,
+        store=True,
+        readonly=False
+    )
+    partner_bill_to_id = fields.Many2one(
+        'res.partner',
+        string="Bill To",
+        compute='_compute_document_partners_details',
+        check_company=True,
+        store=True,
+        readonly=False
+    )
+    partner_ship_from_id = fields.Many2one(
+        'res.partner',
+        string="Dispatch From",
+        compute='_compute_document_partners_details',
+        check_company=True,
+        store=True,
+        readonly=False
+    )
+    partner_ship_to_id = fields.Many2one(
+        'res.partner',
+        string="Ship To",
+        compute='_compute_document_partners_details',
+        check_company=True,
+        store=True,
+        readonly=False
+    )
+
+    # Fields to determine which partner details are editable
+    is_bill_to_editable = fields.Boolean(compute='_compute_is_editable')
+    is_bill_from_editable = fields.Boolean(compute='_compute_is_editable')
+    is_ship_to_editable = fields.Boolean(compute='_compute_is_editable')
+    is_ship_from_editable = fields.Boolean(compute='_compute_is_editable')
+
+    # E-waybill Document Type
+    type_id = fields.Many2one('l10n.in.ewaybill.type', "Document Type", tracking=True, required=True)
+    sub_type_code = fields.Char(related='type_id.sub_type_code')
+
+    # Transportation details
+    distance = fields.Integer("Distance", tracking=True)
+    mode = fields.Selection([
+        ('1', "By Road"),
+        ('2', "Rail"),
+        ('3', "Air"),
+        ('4', "Ship or Ship Cum Road/Rail")
+    ], string="Transportation Mode", copy=False, tracking=True, default='1')
+
+    # Vehicle Number and Type required when transportation mode is By Road.
+    vehicle_no = fields.Char("Vehicle Number", copy=False, tracking=True)
+    vehicle_type = fields.Selection([
+        ('R', "Regular"),
+        ('O', "Over Dimensional Cargo")],
+        string="Vehicle Type",
+        compute='_compute_vehicle_type',
+        store=True,
+        copy=False,
+        tracking=True,
+        readonly=False
+    )
+
+    # Document number and date required in case of transportation mode is Rail, Air or Ship.
+    transportation_doc_no = fields.Char(
+        string="Transporter Doc No",
+        copy=False,
+        tracking=True
+    )
+    transportation_doc_date = fields.Date(
+        string="Transporter Doc Date",
+        copy=False,
+        tracking=True
+    )
+
+    transporter_id = fields.Many2one("res.partner", "Transporter", copy=False, tracking=True)
+
+    error_message = fields.Html(readonly=True)
+    blocking_level = fields.Selection([
+        ("warning", "Warning"),
+        ("error", "Error")],
+        string="Blocking Level", readonly=True)
```

Comment:
I don't think it would make sense here. Because we are not showing error from our end here it's from ewaybill response and also changing blocking level to `actionable_errors` would make data migration more complex